src/adafruit_blinka/microcontroller/mcp2221/mcp2221.py

```diff
@@ -53,9 +53,39 @@ class MCP2221:
     GP_ALT1 = 0b011
     GP_ALT2 = 0b100
 
-    def __init__(self):
-        self._hid = hid.device()
-        self._hid.open(MCP2221.VID, MCP2221.PID)
+    # attempting to reopen the same hid device results in an error,
+    # so we cache and return already opened devices
+    instances = {}
+
+    def __new__(cls, bus_id=None):
+        """Return a (possibly cached) MCP2221 instance."""
+        print(f'MCP2221.__new__; {bus_id=}; instances={list(cls.instances)}')
+        # check for an existing cached instance
+        if bus_id in cls.instances:
+            # return the corresponding cached instance
+            print(f'Returning cached {bus_id}...')
+            return cls.instances[bus_id]
+        if bus_id is None and cls.instances:
+            # return the instance for the first bus_id, if already cached
+            first_bus_id = cls.available_paths(require_mcps=True)[0]
+            if first_bus_id in cls.instances:
+                print(f'Returning cached {first_bus_id}...')
+                return cls.instances[first_bus_id]
```

Comment:
Here I tried to emulate the original behavior: if the user calls `MCP2221()` without the `bus`, it will check if the first device is already opened/cached and return that, if not it will try to open a different one below.

Other possible behaviors are:
* return one of the cached MCP (if any), even if it's not the first
* if no bus is specified, only try to open the first device

Note that simply importing the module will automatically create an instance of the first device, so any subsequent `MCP2221` call (without the `bus`) will return that instance.  This might not happen if another process already opened the first device -- in that case the class will try to open a different one.

Reply:
I now changed this so that:
* `MCP2221()` (with no `bus_id` specified) opens and returns the first available MCP (and raises an error if there are no MCP2221 available).
* `MCP2221(bus_id)` opens and returns the corresponding MCP2221, or return the cached instance if it's already open.